wtransport/src/config.rs

```diff
@@ -388,6 +388,103 @@ impl ServerConfigBuilder<states::WantsIdentity> {
             migration: true,
         })
     }
+
+    /// Configures the server with a custom transport configuration and a default TLS setup
+    /// using the provided [`Identity`].
+    ///
+    /// This method is useful for scenarios where you need to customize the transport settings
```

Comment:
I think we can add a note in the documentation.

```
**See**:  [with_identity] ...
```

mentioning that this method is equivalent to the simpler `with_identity`, but it just allows adding the QUIC transport configuration.

This help the user to understand this behave exactly like `with_identity`.

---

Same comment for the `with_custom_tls` below